engine/datamanager.go

```diff
@@ -856,18 +852,16 @@ func (dm *DataManager) GetActions(key string, skipCache bool, transactionID stri
 	}
 	cache.Set(cachekey, as, cacheCommit(transactionID), transactionID)
 	return
-
 }
 
 func (dm *DataManager) SetActions(key string, as Actions, transactionID string) (err error) {
-	if dm.DataDB().GetStorageType() == utils.MAPSTOR {
-		if err = dm.DataDB().SetActionsDrv(key, as); err != nil {
-			cache.RemKey(utils.ACTION_PREFIX+key, cacheCommit(transactionID), transactionID)
-		}
+	if err = dm.DataDB().SetActionsDrv(key, as); err != nil {
+		return
+	}
+	if err = dm.CacheDataFromDB(utils.ACTION_PREFIX, []string{key}, true); err != nil {
```

Comment:
since this is the last action, you can return directly dm.CacheDataFromDB(utils.ACTION_PREFIX, []string{key}, true)